stdlib/src/builtin/int.mojo

```diff
@@ -1194,6 +1195,64 @@ struct Int(
 
         writer.write(self)
 
+    @staticmethod
+    fn from_bytes[
+        type: DType, big_endian: Bool = False
```

Comment:
I'm in favor of adding a `from_bytes` and `to_bytes` method, but I'm not a fan of using a boolean to represent the endianness here. If we wanted to be Pythonic, we could use `StringLiteral` in the parameter and constrain it to be `"little"` or `"big"`, or we could make an `Endian` wrapper struct which we can parameterize other methods on in the future. WDYT @JoeLoser?

I also don't think we want to add a `DType` parameter here. We can just get the `sizeof[Int]()`, or perhaps add a helper to get the `DType` equivalent to `Int`.

Reply:
I was adding these methods because they are very convenient for networking protocols. For example I am needing these for a pure Mojo implementation of websockets.

If we don't have the `DType` and `big_endian` parameters, and we get an array of 60 bytes from some stream, how can we do the following cases:

1. I want to get 10 UInt64 integers (10x6=60)
2. I want to get 10 Int64 integers
3. I want to get 30 UInt16 integers (30x2=60)
4. I want to get 30 Int16 integers

Python implementation supports both, but as arguments, it's slower as we cannot do it in comptime. But if we are implementing something IO-related, like WebSockets, we already know the endianness of the target we want to read from or write to, so it's faster if we use parameters.

Reply:
If anything, these comments make me think that `from_bytes` and `to_bytes` belongs on `SIMD`, not `Int`. 

> Python implementation of int.as_bytes() have the length and byteorder arguments, and int.from_bytes() assumes we always have a 4 bytes integers and only have the byteorder one, but as arguments (run-time) which is slower and in Mojo we can do it comptime.

I agree :) that's whyi still said it should be a parameter, but my comments were questioning what the parameter should look like, because I'm not sold on `Bool`.

Reply:
Yeah if we follow Python's lead, `bytes` is binary `Byte` and can be encoded in `UInt16` etc. We could also hold off on doing it like Python and let this function be inferred and have `bytes: Span[Scalar[D]]`.

I think `type` shouldn't be used as prolifically as it is for parameters since it's the name of Python's `type` builtin function that I think we'll eventually have in Mojo as well.
```suggestion
        D: DType, big_endian: Bool = False
```

---
> questioning what the parameter should look like, because I'm not sold on Bool.

As for this, I personally never liked `sys.byteorder` returning `"big"` or `"little"` because there is no third option, so a boolean makes sense to me :man_shrugging:. But compatibility may be a solid argument in favor of it.

Reply:
We are already diverging from the Python `int.from_bytes()` and `int.to_bytes()`. As a Python developer, I personally don't mind making a slight adaptation (e.g., `big` -> `True` and `small` -> `False`). The real issue would be if there is no equivalent implementation in Mojo's stdlib; you would need to implement the logic yourself.